ext/openssl/openssl.c

```diff
@@ -808,6 +813,52 @@ static int add_oid_section(struct php_x509_request * req TSRMLS_DC) /* {{{ */
 static const EVP_CIPHER * php_openssl_get_evp_cipher_from_algo(long algo);
 
 
+
+
+
+static PHP_SSL_ERROR_CONTEXT *ssl_error_context;
```

Comment:
This should very likely be a per-thread global. Furthermore it should be destroyed on request shutdown (unless errors like across requests, in which case it should be destroyed on module shutdown).